TensorIO/Classes/TensorFlow/TIOTensorFlowData/NSNumber+TIOTensorFlowData.mm

```diff
@@ -33,37 +33,66 @@ - (nullable instancetype)initWithTensor:(tensorflow::Tensor)tensor description:(
     TIODataDequantizer dequantizer = ((TIOVectorLayerDescription*)description).dequantizer;
     
     if ( description.isQuantized && dequantizer != nil ) {
-        uint8_t value = tensor.scalar<uint8_t>()(0);
+        auto flat_tensor = tensor.flat<uint8_t>();
+        auto tensor_data = flat_tensor.data();
+        uint8_t value = tensor_data[0];
         return [self initWithFloat:dequantizer(value)];
     } else if ( description.isQuantized && dequantizer == nil ) {
-        uint8_t value = tensor.scalar<uint8_t>()(0);
+        auto flat_tensor = tensor.flat<uint8_t>();
+        auto tensor_data = flat_tensor.data();
+        uint8_t value = tensor_data[0];
         return [self initWithUnsignedChar:value];
     } else {
-        float value = tensor.scalar<float>()(0);
+        auto flat_tensor = tensor.flat<float_t>();
+        auto tensor_data = flat_tensor.data();
+        float_t value = tensor_data[0];
         return [self initWithFloat:value];
     }
 }
 
 - (tensorflow::Tensor)tensorWithDescription:(id<TIOLayerDescription>)description {
     assert([description isKindOfClass:TIOVectorLayerDescription.class]);
     
+    // TODO: verify that the shape is either [1] or [-1,1] if batched
```

Comment:
https://github.com/doc-ai/tensorio-ios/issues/41 and https://github.com/doc-ai/tensorio-ios/issues/38